cmd/collector/app/builder/span_handler_builder.go

```diff
@@ -48,157 +46,125 @@ var (
 	errMissingElasticSearchConfig = errors.New("ElasticSearch not configured")
 )
 
-// SpanHandlerBuilder builds span (Jaeger and zipkin) handlers
-type SpanHandlerBuilder interface {
-	BuildHandlers() (app.ZipkinSpansHandler, app.JaegerBatchesHandler, error)
+// SpanHandlerBuilder holds configuration required for handlers
+type SpanHandlerBuilder struct {
+	cOpts *CollectorOptions
+	builder *collectorBuilder
 }
 
-// NewSpanHandlerBuilder returns a span handler
-func NewSpanHandlerBuilder(opts ...basicB.Option) (SpanHandlerBuilder, error) {
+type collectorBuilder struct {
```

Comment:
not sure why you need this struct to be separate from SpanHandlerBuilder, since all 3 times you create it in different functions it has the exact same fields except for the span writer. So instead you can save all those fields on SpanHandlerBuilder here, in the constructor, and the 3 dispatched functions would only need to init the span writer.